src/core/components/subscription_manager.js

```diff
@@ -396,7 +396,7 @@ export default class {
         announce.publisher = message.issuingClientId;
 
         if (this._config.cipherKey) {
-          announce.message = this._crypto.decrypt(message.payload);
+          announce.message = this._crypto.decrypt(message.payload) || this._config.customDecrypt(message.payload, this._config.cipherKey);
```

Comment:
let's drive this further into the `crypto` class instead of augmenting the endpoint file.